src/goto-programs/class_hierarchy.cpp

```diff
@@ -99,3 +104,8 @@ void class_hierarchyt::output(std::ostream &out) const
           << ch << '\n';
   }
 }
+
+bool class_hierarchyt::is_abstract_class(const irep_idt &c) const
+{
+  return abstract_classes.count(c) > 0;
+}
```

Comment:
I might not have a sufficient understanding of this code, but it seems the approach taken here has potential for problems: this function may return different answers depending on _when_ it is called.

I'd strongly recommend doing lookups in the `symbol_table` instead, thus avoiding the need for `abstract_classes` altogether.

Reply:
@tautschnig definitely right with both comments